toptal/picasso
# test(storybook): add optional breakpoints screenshoting

[FX-4089]

### Description

Add `screenshotBreakpoints` to Picasso Storybook, so stories can optionally be screenshotted on the different Picasso breakpoints

### How to test


- [Temploy](https://picasso.toptal.net/fx-4089-add-all-breakpoint-ranges-as-targets-in-visual-tests-storybook)
- Happo should show new screenshots for the [responsive components](https://toptal-core.atlassian.net/wiki/spaces/FE/pages/3300786404/BASE+design+and+Picasso+synchronisation#Estimations-and-scope) that are not being tested on [cypress](https://toptal-core.atlassian.net/browse/FX-4198)

### Screenshots

No screenshots

### Development checks

- [NA] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [NA] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [NA] Annotate all `props` in component with documentation
- [NA] Create `examples` for component
- [NA] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [NA] Covered with tests

**Breaking change**

- [NA] codemod is created and showcased in the changeset
- [NA] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-4089]: https://toptal-core.atlassian.net/browse/FX-4089?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
